File: probe/src/main/java/eu/darken/porter/probe/ProbeService.java
package eu.darken.porter.probe;

import android.os.Process;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ProbeService extends IProbe.Stub {
    @Override public void destroy() { System.exit(0); }

    @Override public int uid() { return Process.myUid(); }

    @Override public String readFile(String path) {
        try (BufferedReader reader = new BufferedReader(new FileReader(path))) {
            return reader.readLine();
        } catch (IOException e) {
            return e.toString();
        }
    }
}

File: probe/src/main/java/eu/darken/porter/probe/ProbeService.kt
package eu.darken.porter.probe

import android.os.Process
import java.io.File
import java.io.IOException
import kotlin.system.exitProcess

class ProbeService : IProbe.Stub() {

    override fun destroy() {
        exitProcess(0)
    }

    override fun uid(): Int = Process.myUid()

    override fun readFile(path: String): String? = try {
        File(path).bufferedReader().use { it.readLine() }
    } catch (e: IOException) {
        e.toString()
    }
}
